### ISSUE TITLE
[Bug]: Bug #31117 NOT FIXED

### ISSUE THREAD

**Issue by dtserekhman-starz** [hashicorp/terraform-provider-aws#33076]
### Terraform Core Version

1.5.5

### AWS Provider Version

5.5.0

### Affected Resource(s)

aws_vpc_endpoint

### Expected Behavior

Original issue was reported in Bug #31117. 

I understand the solution introduced additional attribute to the `aws_vpc_endpoint` as follows:

```terraform
  dns_options {
    private_dns_only_for_inbound_resolver_endpoint = false
  }
```

My particular case is such that I have S3 Interface VPC Endpoint, and I need the settings to keep as follows:

```terraform
  dns_options {
    private_dns_only_for_inbound_resolver_endpoint = false
  }
```

However, I'm still getting error as specified in the next section.

### Actual Behavior

As explained above, I have the following settings set on my `aws_vpc_endpoint` for S3 Interface VPC Endpoint:

```terraform
  private_dns_enabled = true

  dns_options {
    private_dns_only_for_inbound_resolver_endpoint = false
  }
```

I do not have Gateway S3 Endpoint installed in my VPC since I'm keeping `private_dns_only_for_inbound_resolver_endpoint` setting false and don't need to rely on the Gateway endpoint.

However, when I run terraform apply I'm still getting an error as though I've set `private_dns_only_for_inbound_resolver_endpoint` to `true` while it's not the case:

```shell
aws_vpc_endpoint.s3_interface_endpoint: Creating...
╷
│ Error: creating EC2 VPC Endpoint (com.amazonaws.us-west-2.s3): InvalidParameter: To set PrivateDnsOnlyForInboundResolverEndpoint to true, the VPC vpc-0f7536c38326c5ba6 must have a Gateway endpoint for the service.
│       status code: 400, request id: 9c8cb346-729c-49ee-abda-01897566bb00
│ 
│   with aws_vpc_endpoint.s3_interface_endpoint,
│   on _s3_vpc_interface_endpoint.tf line 2, in resource "aws_vpc_endpoint" "s3_interface_endpoint":
│    2: resource "aws_vpc_endpoint" "s3_interface_endpoint" {
│ 
```

### Terraform Configuration Files

```shell
aws_vpc_endpoint.s3_interface_endpoint: Creating...
╷
│ Error: creating EC2 VPC Endpoint (com.amazonaws.us-west-2.s3): InvalidParameter: To set PrivateDnsOnlyForInboundResolverEndpoint to true, the VPC vpc-0f7536c38326c5ba6 must have a Gateway endpoint for the service.
│       status code: 400, request id: 5e0fa4fc-edef-473a-b30d-93c679728277
│ 
│   with aws_vpc_endpoint.s3_interface_endpoint,
│   on _s3_vpc_interface_endpoint.tf line 2, in resource "aws_vpc_endpoint" "s3_interface_endpoint":
│    2: resource "aws_vpc_endpoint" "s3_interface_endpoint" {
│ 
╵
```

### Steps to Reproduce

Please see prior sections

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @dtserekhman-starz 👋 Thank you for taking the time to raise this and let us know that the fix didn't seem to work. In looking into this a bit more, I believe this was actually caught with #32355, which was included in version 5.7.0. Can you test with AWS Provider 5.7.0 or later and let me know if this issue was resolved this time?

**Comment by justinretzolk** [comment]
Since we haven't heard back, I'm going to close this issue. If you're still having trouble, please feel free to open a new issue, referencing this one for context as needed.
